cfme/control/explorer/policies.py

```diff
@@ -344,12 +345,40 @@ def assign_events(self, *events, **kwargs):
         view = navigate_to(self, "Details")
         view.configuration.item_select("Edit this Policy's Event assignments")
         view = self.create_view(EditPolicyEventAssignments, wait="20s")
+        # decide if any events are already assigned
+        events = [event for event in events if not self.is_event_assigned(event)]
+        # select events
         changed = view.fill({"events": events})
         if changed:
             view.save_button.click()
+            view.flash.assert_success_message('Policy "{}" was saved'.format(self.description))
         else:
             view.cancel_button.click()
-        view.flash.assert_success_message('Policy "{}" was saved'.format(self.description))
+            view.flash.assert_success_message(
+                'Edit of Policy "{}" was cancelled by the user'.format(self.description)
+            )
+
+    def unassign_events(self, *events, **kwargs):
+        """Unassign events from this Policy. """
+        events = list(events)
+        extend = kwargs.pop("extend", False)
+        if extend:
```

Comment:
You can put the `kwargs.pop()` call in-line here since we aren't otherwise using `extend`